src/main/scala/org/kframework/kore/interfaces/Interfaces.scala

```diff
@@ -481,61 +481,251 @@ object pattern {
     def unapply(arg: Equals): Option[(Pattern, Pattern)] = Some(arg._1, arg._2)
   }
 
+
 }
 
-/**
-  * Provides a Builder type that allows building Pattern types in [[pattern]].
-  *
-  * Sample Usage -
-  *
-  * Given a concrete implementation of [[build.Builders]], one can create new patters -
-  * {{{
-  *   /* Given Concrete Implementation */
-  *   val builder: Builders = ConcreteBuilders
-  *   /* A Pattern Or(X:Int, Y:Int) can be constructed in the following way */
-  *   val or: Or = builder.Or(builder.Variable("X", Sort("Int")), builder."Y", Sort("Int"))
-  * }}}
-  *
-  */
-object build {
+object outer {
 
-  import pattern._
+  import org.kframework.kore.interfaces.{pattern => p}
+
+  type Attributes = Seq[p.Pattern]
+
+  /**
+    * Trait That is Extended by Every Outer Construct.
+    */
+  trait Outer {
+    val att: Attributes
+    val sorts: Set[p.Sort]
```

Comment:
What are these `sorts`, `symbols` and `sentences` here?  The declared ones, the used ones, the error ones, etc.?  Also, they feel like utilities and not as part of the main KORE interface.  Shouldn't we keep things clean at this stage?  Maybe we should make that separation we were talking about, between a Main KORE interface and an Extended KORE interface, asap.